### PR TITLE
Requirements update

### PR DESCRIPTION
**Pull request by harshbafna**
## Description

Refactor the torch specific requirement files and split them into multiple files based on OS and CUDA version. The enhanced files are designed based on the different pip commands documented in https://pytorch.org/

**Testing done:**

Machine | CPU/GPU | Install Dependency | Install From Src | Sanity Suite | Regression Suite | Remarks
-- | -- | -- | -- | -- | -- | -- |
Mac OS Local | CPU | Pass | Pass | Pass | Pass |  
Mac OS Docker | CPU | Pass | Pass | Pass | Pass | Started   container using root user
Windows 10 Pro (Azure) | CPU | Pass | Pass | Pass | Pass |  
Windows Server2019 | CPU | Pass | Pass | Pass | Pass |  
EC2 Ubuntu Base Image | CPU | Pass | Pass | Pass | Pass |  
EC2 Ubuntu DLAMI | CPU | Pass | Pass | Pass | Pass |  
Ubuntu Docker | CPU | Pass | Pass | Pass | Pass | Started   container using root user
EC2 Ubuntu Base Image | GPU | Pass | Pass | Pass | Pass |  
Windows Server2019 | GPU | Pass | Pass | Pass | Pass |  
EC2 Ubuntu DLAMI CUDA 11.0 | GPU | Pass | Pass | Pass | Pass |  
EC2 Ubuntu DLAMI CUDA 10.2 | GPU | Pass | Pass | Pass | Pass |  
EC2 Ubuntu DLAMI CUDA 10.1 | GPU | Pass | Pass | Pass | Pass |  
Ubuntu Docker | GPU | Pass | Pass | Pass | Pass | Started   container using root user

### CODE REVIEW

**Review by dhanainme — CHANGES_REQUESTED** [review]
Can you please share the details of the testing that was done for 

* CPU - Prod / Dev
* GPU - Prod / Dev
* Windows - Prod / Dev
* Docker / Docker Dev

File: requirements/torch_cu102_linux.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# pip install torch torchvision
```
[the remark this answers is not in the document]
Comment (harshbafna) [reply requirements/torch_cu102_linux.txt:1]: this has been added in every torch related requirements file just as a reference of the official command provided by pytorch.org for installing different torch packages.

For example: `torch_cu102_linux.txt` translates to torch packages for Linux environment with CUDA 10.2 and the official command provided on pytorch.org is 

```
pip install torch torchvision
```

**Comment by harshbafna** [comment]
> Can you please share the details of the testing that was done for
> 
> * CPU - Prod / Dev
> * GPU - Prod / Dev
> * Windows - Prod / Dev
> * Docker / Docker Dev

Added the table for manual testing done on this branch.

**Review by dhanainme — APPROVED** [review]

File: requirements/developer.txt [hunk 2]
```diff
@@ -4,8 +4,8 @@ pytest
 pylint==2.6.0
 pytest-mock
 pytest-cov
-nvgpu
-numpy
+nvgpu; sys_platform != 'win32'
+nvgpu==0.8.0; sys_platform == 'win32'
 grpcio
 protobuf
 grpcio-tools
```
Comment (lxning) [inline requirements/developer.txt:11]: should transformer be set as 4.0.0?

Comment (harshbafna) [reply requirements/developer.txt:11]: @lxning: Transformers is a very BERT model-specific package. This PR targets to stabilize the install_dependency script for different OS/CUDA environment.

Upgrading the transformers version to 4.0.0 will add un-necessary testing and fixes of BERT models. This should be taken up as part of #927

**Review by lxning — APPROVED** [review]

**Pull request merged**